spec/requests/passwords_spec.rb

```diff
@@ -1,17 +1,40 @@
 require 'rails_helper'
 
 RSpec.describe "Passwords", type: :request do
-  describe "GET /edit" do
-    it "returns http success" do
-      get "/passwords/edit"
-      expect(response).to have_http_status(:success)
-    end
+  before do
+    user = create(:user)
+    sign_in user
+    get root_path
   end
 
   describe "GET /update" do
-    it "returns http success" do
-      get "/passwords/update"
-      expect(response).to have_http_status(:success)
+    it "パスワードが変更できること" do
+      put password_path, params: {
+        user: {
+          password: "password1",
+          password_confirmation: "password1",
+          current_password: "password"
+        }
+      }
+      # expect(response.status).to eq 200
+      expect(response.status).to eq 302
+      follow_redirect!
+      expect(response.body).to include "変更"
+    end
+
+    it "パスワードが変更できないこと" do
+      put password_path, params: {
+        user: {
+          password: "pa",
+          password_confirmation: "pass",
+          current_password: "pas"
+        }
+      }
+      expect(response.status).to eq 302
+      follow_redirect!
+      expect(response.body).to include "一致しません"
+      expect(response.body).to include "不正な値です"
+      expect(response.body).to include "パスワードの変更"
     end
```

Comment:
パスワードは更新できていませんが、statusが`200`を返してきてしまいます。
エラーログは下記の表示がされています。
```bash
Passwords
  GET /update
DEPRECATION WARNING: [Devise] bypass option is deprecated and it will be removed in future version of Devise.
Please use bypass_sign_in method instead.
Example:

  bypass_sign_in(user)
 (called from update at /webapp/app/controllers/passwords_controller.rb:7)
    パスワードが変更できること
    パスワードが変更できないこと (FAILED - 1)

Failures:

  1) Passwords GET /update パスワードが変更できないこと
     Failure/Error: expect(response.status).to eq 302

       expected: 302
            got: 200

       (compared using ==)
     # ./spec/requests/passwords_spec.rb:33:in `block (3 levels) in <top (required)>'

Finished in 10.7 seconds (files took 17.16 seconds to load)
2 examples, 1 failure

Failed examples:

rspec ./spec/requests/passwords_spec.rb:25 # Passwords GET /update パスワードが変更できないこと
```

Reply:
> パスワードは更新できていませんが、statusが200を返してきてしまいます。

エラーになって、リダイレクトされないからじゃないですか。 エラーの場合は同じページに遷移するので。